Sources/Apodini/Semantic Model Builder/Model/Endpoint.swift

```diff
@@ -137,19 +127,35 @@ class EndpointsTreeNode {
     private var childContainsPathParameter = false
     
     lazy var absolutePath: [_PathComponent] = {
-        var absolutePath: [_PathComponent] = []
-        collectAbsolutePath(&absolutePath)
-        return absolutePath
+        guard let parent = parent else {
+            return []
+        }
+        
+        return parent.absolutePath + [path]
     }()
     
     lazy var relationships: [EndpointRelationship] = {
-        var relationships: [EndpointRelationship] = []
-        
-        for (name, child) in nodeChildren {
-            child.collectRelationships(name: name, &relationships)
+        func collectRelationships(_ node: EndpointsTreeNode, name: String) -> [EndpointRelationship] {
```

Comment:
Any reason for nesting the function into a lazy property? Feels a bit unclean to me.